### PR TITLE
[java-client][java-jaxrs-server][jackson] Add @JsonPropertyOrder to models

### LINKED ISSUE #3629: [REQ][java][jackson] add @JsonPropertyOrder annotation
When the model classes are serialized to JSON the order of the JSON members are not controlled.

Given this schema:

```yaml
components:
  schemas:
    SomeObj:
      type: object
      properties:
        id:
          type: integer
          format: int64
        firstName:
          type: string
        lastName:
          type: string
```

This is not a big deal because the order is not relevant in a JSON object:

```json
{
"id" : 42,
"firstName": "John",
"lastName": "Doe"
}
```

Is the same than:
```json
{
"lastName": "Doe",
"id" : 42,
"firstName": "John"
}
```

But if the JSON objects are displayed or stored as file, it would be nice to have always the same order (the same as in the OpenAPI specification).

With Jackson the solution is to use the `@JsonPropertyOrder` property.

In my opinion this should be generated (maybe with an option to hide it for user that do not want it).

### PR DESCRIPTION
**Pull request by jmini**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (xor instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language. Java @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04)

### Description of the PR

Fixes #3629

As discussed with the @OpenAPITools/generator-core-team, we do not add a new flag for this.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/JavaJaxRS/pojo.mustache [hunk 1]
```diff
@@ -1,6 +1,13 @@
 /**
  * {{#description}}{{.}}{{/description}}{{^description}}{{classname}}{{/description}}
  */{{#description}}
+{{#jackson}}
+@JsonPropertyOrder({
+{{#vars}}
+  {{classname}}.JSON_PROPERTY_{{nameInSnakeCase}},
```
Comment (cbornet) [inline modules/openapi-generator/src/main/resources/JavaJaxRS/pojo.mustache]: Maybe don't output the `,'  for the last var ?

Comment (jmini) [reply modules/openapi-generator/src/main/resources/JavaJaxRS/pojo.mustache]: It compiles, but is is true that it is not nice. Thank you for the input!

**Comment by jmini** [comment]
@cbornet I did the requested change.

**Comment by cbornet** [comment]
:+1:

**Comment by rockbird** [comment]
Can this be added to Java Spring generator too ?

**Comment by aitia** [comment]
Can this be added to Java Spring generator too please ?

**Pull request merged**